### PR TITLE
Da/unencrypted no refinery

### PR DESCRIPTION
**Pull request by danamansana**
Removes refinery from the app. Currently we are setting the google api endpoint by hand in AWS, changing this will be the subject of a future PR.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -67,6 +68,7 @@ const pageTitle = req.originalUrl === '/' ? 'Search NYPL.org' : "Search Results
     appEnv: process.env.APP_ENV,
```
[the remark this answers is not in the document]
Comment (danamansana) [reply server.js]: Fixed

File: src/app/utils/SearchModel.js [hunk 2]
```diff
@@ -18,13 +18,10 @@ import {
 const fetchResultLength = (data) => {
   try {
     const {
-      attributes: {
-        meta: {
-          'total-results': totalResults = 0,
-        },
+      searchInformation: {
+        formattedTotalResults: totalResults,
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/app/utils/SearchModel.js]: Fixed

File: src/app/utils/SearchModel.js [hunk 3]
```diff
@@ -69,6 +66,8 @@ const extractSearchElements = (requestCombo) => {
   };
 };
 
+const displayName = (name) => name.split("_").map(word => word[0].toUpperCase() + word.slice(1)).join(" ")
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/app/utils/SearchModel.js]: Good catch!

File: src/app/utils/SearchModel.js [hunk 4]
```diff
@@ -86,35 +85,36 @@ const extractSearchElements = (requestCombo) => {
...
     return '';
   }
 
   const displayNameArray = _map(labelsArray, (label) => {
+    console.log('labels: ', labelsArray)
     if (!label.displayName || !label.name) {
       return {
-        name: '',
-        displayName: '',
+        name: ''
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/app/utils/SearchModel.js]: Nice

File: src/app/utils/SearchModel.js [hunk 5]
```diff
@@ -157,13 +157,29 @@ const secureHttpsProtocol = (url) =>
  * @return {String}
  */
 const fetchItemFeature = (item, feature) => {
-  if (!item.attributes[feature]) {
+  if (!item[feature]) {
     return '';
   }
 
-  return item.attributes[feature];
+  return item[feature];
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/app/utils/SearchModel.js]: Nice!

File: src/app/utils/SearchModel.js [hunk 6]
```diff
@@ -157,13 +157,29 @@ const secureHttpsProtocol = (url) =>
...
 };
 
+const getNestedData = (array, item) => {
+  return array.reduce((acc, el) => acc[el] ? acc[el] : '', item)
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/app/utils/SearchModel.js]: Yes

File: src/app/utils/SearchModel.js [hunk 7]
```diff
@@ -157,13 +157,29 @@ const secureHttpsProtocol = (url) =>
...
+}
+
+const thumbnailLogging = (item) => {
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/app/utils/SearchModel.js]: Good idea

File: src/app/utils/SearchModel.js [hunk 8]
```diff
@@ -179,7 +195,8 @@ const fetchItemFeature = (item, feature) => {
  * @return {Object}
  */
 const fetchItem = (item, searchRequest) => {
-  if (!item || !item.attributes) {
+  console.log('item: ', item)
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/app/utils/SearchModel.js]: Fixed

File: src/server/ApiRoutes/ApiRoutes.js [hunk 9]
```diff
@@ -15,39 +15,38 @@ const { api, searchApi } = appConfig;
 
 const router = express.Router();
 const appEnvironment = process.env.APP_ENV || 'production';
-const apiRoot = api.root[appEnvironment];
+// const apiRoot = api.root[appEnvironment];
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/server/ApiRoutes/ApiRoutes.js]: Fixed

File: src/server/ApiRoutes/ApiRoutes.js [hunk 10]
```diff
@@ -79,31 +77,26 @@ const requestSearchResult = (req, res, next) => {
...
   if (!req.query.start) {
     res.json({});
     return;
   }
-
+  
   getSearchData(searchApiUrl)
     .then((searchData) => {
-      const searchParsed = parser.parse(searchData.data, searchOptions);
+      const data = searchData.data;
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/server/ApiRoutes/ApiRoutes.js]: Let's leave it for now

File: src/server/ApiRoutes/ApiRoutes.js [hunk 11]
```diff
@@ -121,12 +114,12 @@ router
 
 // For reverse proxy URLs
 router
-  .route('/:searchKeyword/:searchFilter?')
+  .route('/:searchRequest/:searchFilter?')
```
[the remark this answers is not in the document]
Comment (danamansana) [reply src/server/ApiRoutes/ApiRoutes.js]: This allows us to generate the url with the same method for both routes

File: test/SearchModel.test.js [hunk 12]
```diff
@@ -71,7 +65,7 @@ describe('fetchResultItems', () => {
     }
   );
 });
-
+//
```
[the remark this answers is not in the document]
Comment (danamansana) [reply test/SearchModel.test.js]: Fixed

**Comment by ktp242** [comment]
Sorry I didn't get time to review this! It will be my first thing tmr.

File: server.js [hunk 13]
```diff
@@ -67,6 +68,7 @@ const pageTitle = req.originalUrl === '/' ? 'Search NYPL.org' : "Search Results
...
     apiUrl: (res.locals.data) ? res.locals.data.completeApiUrl : '',
     isProduction,
+    apiRoot
```
Comment (ktp242) [inline server.js]: I don't think we are using it on client side. We can remove this.

File: src/app/utils/SearchModel.js [hunk 14]
```diff
@@ -69,6 +66,8 @@ const extractSearchElements = (requestCombo) => {
   };
 };
 
+const displayName = (name) => name.split("_").map(word => word ? word[0].toUpperCase() + word.slice(1) : '').join(" ")
```
Comment (ktp242) [inline src/app/utils/SearchModel.js]: If we have a input as "a_", the function will return the result with a trailing space in the end, you can use trim() to remove it.

If you are comfortable with Regular Expression, you can use it to get the same result and save you from using the loop. But this function works just fine as well!

**Pull request merged**